homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -42,66 +44,91 @@
-        see(
-            mac="{}{}".format(BT_PREFIX, mac),
-            host_name=name,
-            attributes=attributes,
-            source_type=SOURCE_TYPE_BLUETOOTH,
-        )
-
-    device_id = config.get(CONF_DEVICE_ID)
-
-    def discover_devices():
-        """Discover Bluetooth devices."""
-        result = bluetooth.discover_devices(
-            duration=8,
-            lookup_names=True,
-            flush_cache=True,
-            lookup_class=False,
-            device_id=device_id,
-        )
-        _LOGGER.debug("Bluetooth devices discovered = %d", len(result))
-        return result
-
-    yaml_path = hass.config.path(YAML_DEVICES)
-    devs_to_track = []
-    devs_donot_track = []
-
-    # Load all known devices.
-    # We just need the devices so set consider_home and home range
-    # to 0
+def is_bluetooth_device(device) -> bool:
+    """Check whether a device is a bluetooth device by its mac."""
+    return device.mac and device.mac[:3].upper() == BT_PREFIX
+
+
+def discover_devices(bluetooth, device_id: int) -> List[Tuple[str, str]]:
+    """Discover Bluetooth devices."""
+    result = bluetooth.discover_devices(
+        duration=8,
+        lookup_names=True,
+        flush_cache=True,
+        lookup_class=False,
+        device_id=device_id,
+    )
+    _LOGGER.debug("Bluetooth devices discovered = %d", len(result))
+    return result
+
+
+def see_device(see, mac: str, device_name: str, rssi=None) -> None:
+    """Mark a device as seen."""
+    attributes = {}
+    if rssi is not None:
+        attributes["rssi"] = rssi
+    see(
+        mac="{}{}".format(BT_PREFIX, mac),
+        host_name=device_name,
+        attributes=attributes,
+        source_type=SOURCE_TYPE_BLUETOOTH,
+    )
+
+
+def get_tracking_devices(hass: HomeAssistantType) -> Tuple[Set[str], Set[str]]:
+    """
+    Load all known devices.
+
+    We just need the devices so set consider_home and home range to 0
+    """
+    yaml_path: str = hass.config.path(YAML_DEVICES)
+    devices_to_track: Set[str] = set()
+    devices_to_not_track: Set[str] = set()
+
     for device in run_coroutine_threadsafe(
         async_load_config(yaml_path, hass, 0), hass.loop
     ).result():
         # Check if device is a valid bluetooth device
-        if device.mac and device.mac[:3].upper() == BT_PREFIX:
-            if device.track:
-                devs_to_track.append(device.mac[3:])
-            else:
-                devs_donot_track.append(device.mac[3:])
+        if not is_bluetooth_device(device):
+            continue
+
+        normalized_mac: str = device.mac[3:]
+        if device.track:
+            devices_to_track.add(normalized_mac)
+        else:
+            devices_to_not_track.add(normalized_mac)
+
+    return devices_to_track, devices_to_not_track
+
+
+def setup_scanner(hass: HomeAssistantType, config: dict, see, discovery_info=None):
+    """Set up the Bluetooth Scanner."""
+    # pylint: disable=import-error
+    import bluetooth
```

Comment:
3rd party imports can be moved to the top of the module.

Reply:
:+1: